### ISSUE TITLE
CS1070 with incorrect field with local function

### ISSUE THREAD

**Issue by chm-tm** [dotnet/roslyn#35578]
**Version Used**: 
- 2.9.0
- 2.10.0-beta2-63501-03+b9fb1610c87cccc8ceb74a770dba261a58e39c4a
- [d79e5f](https://github.com/dotnet/roslyn/commit/d79e5f2ce4d2d74d2c34afdfe40953d9c7caf69a)

**Steps to Reproduce**:
Let the following code be analyzed:
```C#
using System.Threading;					
public class Program
{
	static void Foo()
	{
		CancellationTokenRegistration ctr;
		void h() => ctr.Dispose();
		h();
	}
}
```
**Expected Behavior**:
error CS0170: Use of possibly unassigned field 'ctr'
in the body of the local function

**Actual Behavior**:
error CS0170: Use of possibly unassigned field 'm_index'
where h is called

**Comment by svick** [comment]
I believe you are right that the error message is incorrect. (Though the correct error is "CS0165: Use of unassigned local variable 'ctr'".) This is slightly complicated by the fact that for value types like `CancellationTokenRegistration`, the compiler does track the assignment status of its fields. But in the end, I think that doesn't matter here and that the compiler should report the whole variable as unassigned, [just like it does if you invoke the method directly in the body of the `Foo` method](https://sharplab.io/#v2:C4LgbgpgTglgZgTwD4AEAMACFBGAdAFQAsoIBDAExgDsBzAbgEgmmBYAKBQGYsAmLbAOzsA3uwY4AbFgAsGAGIB7BQAoAlGNFsmAYVJUAxhAA2R0sBgKq+BQGsIVAEoQaMAM7AoZi1Qz6PdMQY/KFwAETcABwVXCDUArQBfdgSgA).

Though I also looked at the spec, attempting to find the place that says that method invocations on structs require full definite assignment and was left confused. [The spec says](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/variables#definite-assignment):

>  A variable must be definitely assigned at each location where its value is obtained. This ensures that undefined values never occur. The occurrence of a variable in an expression is considered to obtain the value of the variable, except when
>    * the variable is the left operand of a simple assignment,
>    * the variable is passed as an output parameter, or
>    * the variable is a *struct_type* variable and occurs as the left operand of a member access.

Here, the third point should apply, because `ctr.Dispose` is a member access and I couldn't find anything that would override this for invocations on that member access. Does this mean that the spec says that `ctr` does not have to be definitely assigned in the expression `ctr.Dispose()`?
